src/components/ModalWithForm.js

```diff
@@ -8,6 +8,9 @@ export default class ModalWithForm extends Modal {
     this._inputElements = Array.from(
       this.modalForm.querySelectorAll(".modal__input")
     );
+    this._modalSubmitButton = this._modalElement.querySelector(
+      ".modal__submit-button"
+    );
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`